pydp/algorithms/algorithm.py

```diff
@@ -0,0 +1,41 @@
+from functools import wraps
+
+from .._pydp import _algorithms
+
+
+class Algorithm:
+
+    # Class variables
+    __methods_to_wrap = [
+        "epsilon",
+        "privacy_budget_left",
+        "memory_used",
+        "add_entries",
+        "add_entry",
+        "result",
+        "partial_result",
+        "consume_privacy_budget",
```

Comment:
Typo:
we need to remove `consume_private_budget` as it has been removed from the builder.

The core lib says it `// Strictly reduces privacy budget, so is safe to make public.` I guess we need to see if it's desired to expose it.

Either way they need to match up.

Reply:
The line about consume_private_budget that I deleted previously was simply because it was defined again a couple of lines down (basically I removed it from the  *getters* group as it is not a getter).

In any case, I was just exposing the public methods of algorithm.h by default. Indeed, if it is only meant to be used internally and it makes no sense for the user to call it, we shouldn't expose it.

Reply:
All clear now. I already deleted it both from the Python API and the C++ binding.